cheetah/particles.py

```diff
@@ -238,7 +238,7 @@ def __init__(self, mu, cov, energy):
         self._mu = mu
         self._cov = cov
         self.energy = energy
-        if element.device == "auto" and torch.cuda.is_available():
+        if element.device == "cuda":
```

Comment:
We talked about this. I think it's better to check in `__call__` if the beam's device matches the segment's device and issue a meaningful exception when it doesn't.